### ISSUE TITLE
AOP synthetic warnings in java8

### ISSUE THREAD

**Issue by tomislavr** [mybatis/guice#71]
Noticed this issue when having `@Transactional` annotation on methods of repository classes that implement some interfaces with generics:

https://groups.google.com/forum/#!topic/google-guice/-DH5fBD7M30

Did this in my module, the problem is gone and everything looks OK:

```
...
protected static final AbstractMatcher<Method> SYNTHETIC = new AbstractMatcher<Method>() {
    @Override
    public boolean matches(Method method) {
        return method.isSynthetic();
    }
};

...
@Override
protected void bindTransactionInterceptors() 
{
  TransactionalMethodInterceptor interceptor = new TransactionalMethodInterceptor();
  requestInjection(interceptor);
  bindInterceptor(any(), not(SYNTHETIC).and(not(DECLARED_BY_OBJECT).and(annotatedWith(Transactional.class))), interceptor);

  // Intercept classes annotated with Transactional, but avoid "double"
  // interception when a mathod is also annotated inside an annotated
  // class.
  bindInterceptor(annotatedWith(Transactional.class), 
      not(SYNTHETIC).and(not(DECLARED_BY_OBJECT).and(not(annotatedWith(Transactional.class)))), interceptor);
}
```

If this is ok, I suppose `bindTransactionInterceptors()` can implement this change in mybatis-guice?

**Comment by christianpoitras** [comment]
This seems like a reasonable change.

**Comment by christianpoitras** [comment]
Fixed in commit 727e9d727f0166d484e60e50739982ec04e7b6a4.
